models/adlink_test.go

```diff
@@ -0,0 +1,157 @@
+	"go-google-scraper-challenge/models"
+	"go-google-scraper-challenge/models/adlinks"
+	. "go-google-scraper-challenge/tests/helpers"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("AdLink", func() {
+	Describe("CreateAdLink", func() {
+		Context("given ad link with valid params", func() {
+			It("returns the ad link ID", func() {
+				user := FabricateUser("dev@nimblehq.co", "password")
+				result := FabricateResult(user)
+				adLink := &models.AdLink{
+					Result:   result,
+					Position: adlinks.Top,
+					Type:     adlinks.Link,
+				}
+				adLinkID, err := models.CreateAdLink(adLink)
+				if err != nil {
+					Fail("Failed to add result: " + err.Error())
+				}
+
+				Expect(adLinkID).To(BeNumerically(">", 0))
+			})
+
+			It("returns NO error", func() {
+				user := FabricateUser("dev@nimblehq.co", "password")
+				result := FabricateResult(user)
+				adLink := &models.AdLink{
+					Result:   result,
+					Position: adlinks.Top,
+					Type:     adlinks.Link,
+				}
+				_, err := models.CreateAdLink(adLink)
+
+				Expect(err).To(BeNil())
+			})
+		})
+
+		Context("given adLink with INVALID params", func() {
+			Context("given NO user and keyword", func() {
+				It("returns an error", func() {
+					adLink := &models.AdLink{}
+
+					adLinkID, err := models.CreateAdLink(adLink)
+
+					Expect(err.Error()).To(Equal("field `go-google-scraper-challenge/models.AdLink.Result` cannot be NULL"))
+					Expect(adLinkID).To(Equal(int64(0)))
+				})
+			})
+		})
+	})
+
+	Describe("#GetAdLinkById", func() {
+		Context("given adLink id exist in the system", func() {
+			It("returns adLink with given id", func() {
+				user := FabricateUser("dev@nimblehq.co", "password")
+				result := FabricateResult(user)
+				existAdLink := FabricateAdLink(result)
+				adLink, err := models.GetAdLinkById(existAdLink.Id)
+				if err != nil {
+					Fail("Failed to get adLink with ID")
+				}
+
+				Expect(adLink.Link).To(Equal(existAdLink.Link))
+				Expect(adLink.Result.Id).To(Equal(result.Id))
+			})
+		})
+
+		Context("given adLink id does NOT exist in the system", func() {
+			It("returns false", func() {
+				adLink, err := models.GetAdLinkById(999)
+
+				Expect(err.Error()).To(ContainSubstring("no row found"))
+				Expect(adLink).To(BeNil())
+			})
+		})
+	})
+
+	Describe("#GetAdLinksByResultId", func() {
+		Context("given a valid result id", func() {
+			It("returns adlinks with the given result id", func() {
+				user := FabricateUser("dev@nimblehq.co", "password")
+				otherUser := FabricateUser("dev2@nimblehq.co", "password")
+				result := FabricateResult(user)
+				otherResult := FabricateResult(otherUser)
+				adLink1 := FabricateAdLink(result)
+				adLink2 := FabricateAdLink(result)
+				otherAdLink := FabricateAdLink(otherResult)
+
+				adLinks, err := models.GetAdLinksByResultId(result.Id)
+				if err != nil {
+					Fail("Failed to get adlinks with Result Id")
+				}
+
+				AdLinkIds := []int64{}
```

Comment:
This is our convention
> Prefer var s []string over s := []string{} when declaring an empty slice.

```suggestion
				var AdLinkIds []int64
```

Please check other places too.

Reply:
@gutakk fixed in [cce647b](https://github.com/carryall/go-google-scraper-challenge/pull/42/commits/cce647b0b55a20c8242f0c8e39f4d2c9c3a786c5)